jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -691,10 +692,13 @@ ecma_set_typedarray_element (ecma_typedarray_info_t *info_p,
     }
   }
 
-  if (ecma_arraybuffer_is_detached (info_p->array_buffer_p))
+  if (ecma_object_class_is (info_p->array_buffer_p, ECMA_OBJECT_CLASS_ARRAY_BUFFER))
   {
-    ecma_free_value (to_num);
-    return ecma_raise_type_error (ECMA_ERR_MSG (ecma_error_arraybuffer_is_detached));
+    if (ecma_arraybuffer_is_detached (info_p->array_buffer_p))
```

Comment:
Use `&&` instead of double `if`-s. By the way, isn't `ecma_arraybuffer_is_detached` always returns false for shared array buffers? This could simplify the code (the original can be kept).